prboom2/src/g_game.c

```diff
@@ -1658,7 +1671,13 @@ void G_DoCompleted (void)
 		    for (i = 0; i < MAXPLAYERS; i++)
 		      players[i].didsecret = false;
 		  }
+	  }
+
+	  if (next[0] || intermission)
+	  {
 		  wminfo.didsecret = players[consoleplayer].didsecret;
+		  wminfo.partime = gamemapinfo->partime;
```

Comment:
There's no guarantee a par time is set, is there? Doesn't this need a check if the par time is nonzero?

Reply:
> There's no guarantee a par time is set, is there? Doesn't this need a check if the par time is nonzero?

The former code didn't check for this either. We need to have consistent implementations across ports, that's why I reverted nearly all of the previous fix.

Reply:
NB: It did check in the part that displays the par time, but this is not to question here.

Reply:
Isn't that going to change the behavior in both cases though?

If no par time is set, a 0-time patch would be drawn, wouldn't it? Before it would just be blank.

Reply:
Good catch! I will change it to
```c
if (wminfo.partime > 0)
  um_pars = true
```